src/plots/cartesian/axis_defaults.js

```diff
@@ -117,5 +121,68 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, coerce,
         }
     }
 
+    // TODO
+    // - does this make sense for 'log', 'category' and 'multicategory' axis types ??
+
+    var breaks = containerIn.breaks;
+    if(Array.isArray(breaks) && breaks.length) {
+        handleArrayContainerDefaults(containerIn, containerOut, {
+            name: 'breaks',
+            inclusionAttr: 'enabled',
+            handleItemDefaults: breaksDefaults
+        });
+        setConvert(containerOut, layoutOut);
+    }
+
     return containerOut;
 };
+
+function breaksDefaults(itemIn, itemOut, containerOut) {
+    function coerce(attr, dflt) {
+        return Lib.coerce(itemIn, itemOut, layoutAttributes.breaks, attr, dflt);
+    }
+
+    var enabled = coerce('enabled');
+
+    if(enabled) {
+        var isDateAxis = containerOut.type === 'date';
+
+        var bnds = coerce('bounds');
+
+        if(bnds && bnds.length >= 2) {
+            if(bnds.length > 2) {
+                itemOut.bounds = itemOut.bounds.slice(0, 2);
+            }
+
+            if(containerOut.autorange === false) {
+                var rng = containerOut.range;
+
+                // if bounds are bigger than the (set) range, disable break
+                if(rng[0] < rng[1]) {
+                    if(bnds[0] < rng[0] && bnds[1] > rng[1]) {
+                        itemOut.enabled = false;
+                        return;
+                    }
+                } else if(bnds[0] > rng[0] && bnds[1] < rng[1]) {
+                    itemOut.enabled = false;
+                    return;
+                }
+            }
```

Comment:
What about handling `autorange: 'reversed'` case?
In the following demos I tried to flip the axes which it didn't work.
https://codepen.io/MojtabaSamimi/pen/yLNzLra?editors=0010
https://codepen.io/MojtabaSamimi/pen/bGdoGyG?editors=0010
https://codepen.io/MojtabaSamimi/pen/QWbqWRj?editors=0010

Reply:
Ah right, I forgot to test those. They won't be handled here as this logic has to do with set (i.e. `autorange: false` ranges, but yeah I'll get this fixed. Thanks!